src/AutoRest.CSharp.V3/Output/Models/Types/ObjectType.cs

```diff
@@ -50,18 +55,62 @@ public ObjectType(ObjectSchema objectSchema, BuildContext context)
 
         public Schema Schema { get; }
 
-        public CSharpType? Inherits => _inheritsType ?? CreateInheritedType();
+        public CSharpType? Inherits => _inheritsType ??= CreateInheritedType();
 
         public ObjectSerialization[] Serializations => _serializations ??= BuildSerializations();
 
         public ObjectTypeProperty[] Properties => _properties ??= BuildProperties().ToArray();
 
         public ObjectTypeDiscriminator? Discriminator => _discriminator ??= BuildDiscriminator();
 
-        public CSharpType? ImplementsDictionaryElementType => _inheritsType ??= CreateInheritedDictionaryElementType();
+        public CSharpType? ImplementsDictionaryElementType => _implementsDictionaryType ??= CreateInheritedDictionaryElementType();
 
         public CSharpType Type { get; }
 
+        public ObjectTypeProperty GetPropertyForSchemaProperty(Property property, bool includeParents = false)
+        {
+            if (!TryGetPropertyForSchemaProperty(p => p.SchemaProperty == property, out ObjectTypeProperty? objectProperty, includeParents))
+            {
+                throw new InvalidOperationException($"Unable to find object property for schema property {property.SerializedName}");
```

Comment:
Can we include the type in the error message?  In all the error messages you're adding, actually?  Unless SerializedName is fully qualified in which case ignore this.